File: web/src/api/auth.api.ts
import { api } from './client';
import type { User } from './types';

export interface AuthResponse {
  accessToken: string;
  refreshToken: string;
  user: User;
}

export const authApi = {
  login: (email: string, password: string) =>
    api.post<AuthResponse>('/auth/login', { email, password }).then((r) => r.data),
  register: (body: { email: string; password: string; fullName: string }) =>
    api.post<AuthResponse>('/auth/register', body).then((r) => r.data),
  me: () => api.get<User>('/users/me').then((r) => r.data),
};

File: mobile/lib/core/api/auth_api.dart
import 'api_client.dart';
import '../storage/token_storage.dart';

class AuthApi {
  final _dio = ApiClient.instance.dio;

  Future<Map<String, dynamic>> login(String email, String password) async {
    final res = await _dio.post('/auth/login', data: {
      'email': email,
      'password': password,
    });
    final data = Map<String, dynamic>.from(res.data);
    await TokenStorage.instance.save(data['accessToken'], data['refreshToken']);
    return Map<String, dynamic>.from(data['user']);
  }

  Future<Map<String, dynamic>> register({
    required String email,
    required String password,
    required String fullName,
  }) async {
    final res = await _dio.post('/auth/register', data: {
      'email': email,
      'password': password,
      'fullName': fullName,
    });
    final data = Map<String, dynamic>.from(res.data);
    await TokenStorage.instance.save(data['accessToken'], data['refreshToken']);
    return Map<String, dynamic>.from(data['user']);
  }

  Future<void> logout() => TokenStorage.instance.clear();
}
